pkg/operator/controller/ingress/status.go

```diff
@@ -66,6 +68,7 @@ func (r *reconciler) syncIngressControllerStatus(ic *operatorv1.IngressControlle
 	degradedCondition, err := computeIngressDegradedCondition(updated.Status.Conditions, updated.Name)
 	errs = append(errs, err)
 	updated.Status.Conditions = MergeConditions(updated.Status.Conditions, degradedCondition)
+	updated.Status.Conditions = MergeConditions(updated.Status.Conditions, computeIngressUpgradeableCondition(ic, deploymentRef, service, infraConfig.Status.PlatformStatus))
```

Comment:
Oof, I missed this during review—`infraConfig.Status` could be nil.  That's why we use the `GetPlatformStatus` helper.  I'll fix this in my backport of #710.

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/711/commits/654c1a9f8a80cfcee59ecbe645764a292f0d0e5c

Reply:
On further investigation, `infraConfig.Status` cannot be nil, but `infraConfig.Status.PlatformStatus` can be, and the consequence of passing a nil value for `plsatform` to `computeIngressUpgradeableCondition` would be the following:

* `computeIngressUpgradeableCondition` passes the nil value to `loadBalancerServiceIsUpgradeable`.
* `loadBalancerServiceIsUpgradeable` passes the nil value to `desiredLoadBalancerService`.
* `desiredLoadBalancerService` passes the nil value to `IsProxyProtocolNeeded`.
* `IsProxyProtocolNeeded` checks for nil and returns an error value.
* The error value gets returned up through `desiredLoadBalancerService` and `loadBalancerServiceIsUpgradeable` to `computeIngressUpgradeableCondition`.
* `computeIngressUpgradeableCondition` reports `Upgradeable=False`.

Thus, if my analysis is correct, this issue will inappropriately block upgrades on born-as-4.1 clusters.

Reply:
@wking pointed https://github.com/openshift/cluster-config-operator/pull/174 out to me; this change in cluster-config-operator should prevent this issue with born-as-4.1 clusters.